DigitalLearningSolutions.Data/Services/ActionPlanService.cs

```diff
@@ -102,6 +102,11 @@ public void AddResourceToActionPlan(int competencyLearningResourceId, int delega
 
         public async Task<IEnumerable<ActionPlanResource>> GetIncompleteActionPlanResources(int delegateId)
         {
+            if (!config.IsSignpostingUsed())
+            {
+                return new List<ActionPlanResource>();
+            }

```

Comment:
Is there a strong preference to put the gate in here rather than where this method is used? It feels like this method shouldn't be being called if signposting isn't enabled.

I think it's worth updating the name of the method if you're putting the gate in here. I generally wouldn't expect a service method to conk out if a config flag is set unless it explicitly says so.